sympy/stats/rv.py

```diff
@@ -32,6 +33,15 @@
 
 x = Symbol('x')
 
+@singledispatch
+def is_random(x):
+    return False
+
+@is_random.register(Boolean)
```

Comment:
why `Boolean`?

Reply:
For expressions of And, Or, Le, etc. that contains rvs.

Reply:
I think `.args` will work for any child of `Basic`. So, probably not needed.

Reply:
I will change that to Basic